crypto/txscript/src/opcodes/mod.rs

```diff
@@ -883,96 +872,134 @@ opcode_list! {
+                _ => Err(TxScriptError::InvalidSource("OpInputCount only applies to transaction inputs".to_string()))
             }
         } else {
             Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
         }
     }
-    opcode OpInputAmount<0xb3, 1>(self, vm) {
+    opcode OpTxOutputCount<0xb4, 1>(self, vm) {
         if vm.kip10_enabled {
             match vm.script_source {
-                ScriptSource::TxInput{
-                    utxo_entry: kaspa_consensus_core::tx::UtxoEntry{
-                        amount,
-                        ..
-                    },
-                    ..
-                } => push_number(*amount as i64, vm),
+                ScriptSource::TxInput{tx, ..} => {
+                    push_number(tx.outputs().len() as i64, vm)
+                },
+                _ => Err(TxScriptError::InvalidSource("OpOutputCount only applies to transaction inputs".to_string()))
+            }
+        } else {
+            Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
+        }
+    }
+    opcode OpTxLockTime<0xb5, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxSubnetId<0xb6, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxGas<0xb7, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxPayload<0xb8, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxId<0xb9, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    // Input related opcodes (following TransactionInput struct field order)
+    opcode OpOutpointTxHash<0xba, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpOutpointTxIdx<0xbb, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    opcode OpTxInputIndex<0xbc, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{id, ..} => {
+                    push_number(id as i64, vm)
+                },
+                _ => Err(TxScriptError::InvalidSource("OpInputIndex only applies to transaction inputs".to_string()))
+            }
+        } else {
+            Err(TxScriptError::InvalidOpcode(format!("{self:?}")))
+        }
+    }
+    opcode OpTxInputSeq<0xbd, 1>(self, vm) Err(TxScriptError::OpcodeReserved(format!("{self:?}")))
+    // UTXO related opcodes (following UtxoEntry struct field order)
+    opcode OpTxInputAmount<0xbe, 1>(self, vm) {
+        if vm.kip10_enabled {
+            match vm.script_source {
+                ScriptSource::TxInput{tx, ..} => {
+                    let [idx]: [i32; 1] = vm.dstack.pop_items()?;
+                    if !(0..=u8::MAX as i32).contains(&idx) {
+                        return Err(TxScriptError::InvalidIndex(idx as usize, tx.inputs().len()))
+                    }
+                    let idx = idx as usize;
```

Comment:
I think we should use `try_into` and make script error otherwise, to avoid negative values etc (applies to the rest of the file as well)

Reply:
commit 046308f0a94592dc39e388ec33eaf90d2931c03b